### ISSUE TITLE
Datadog Agent v6.14.1 crashes  with OOMkilled 

### ISSUE THREAD

**Issue by apostoa** [DataDog/datadog-agent#4468]
**Output of the info page (if this is a bug)**
```
(Paste the output of the info page here)
```

**Describe what happened:**
We noticed that one of the *datadog-agents* it crashes constantly with ```OOMkilled``` issue in our development environment. Also, we've noticed that agent is running on . the same node as *kube-state-metrics*.

**Describe what you expected:**
The actions we took were:
- upgraded the *kube-state-metrics* from *0.8.1* to the latest version  *2.4.1* 
- upgrade Datadog from *6.4.2* to *6.14.1*
- increased the memory limit for datadog-agents 
      Limits:
      cpu:     1500m
      memory:  2560Mi
    Requests:
      cpu:     150m
      memory:  400Mi

The Grafana dashboard do not indicated that the memory limit was reached:

![image](https://user-images.githubusercontent.com/35599450/69276640-84b49700-0be7-11ea-84c6-6f8dff63f086.png)

More details here (logs): https://gist.github.com/apostoa/3bc772ee2b7528af30fbc354bf00e449#file-describe-datadog-agent

**Steps to reproduce the issue:**

**Additional environment details (Operating System, Cloud provider, etc):**
Cloud provider AWS
Kubernetes version : 1.13.12

**Comment by apostoa** [comment]
any ideas @CharlyF ?

**Comment by acim** [comment]
I have read at some other issue that memory limit should be increased if you have enabled  kubernetes_states metrics as well. I am waiting for Datadog support to suggest which memory limit value should be actually used.

**Comment by dmizelle** [comment]
Hey @acim @apostoa 

Are y'all still experiencing this? I ran into the same issue and am working with Datadog support. 

I've given them a few dumps from pprof (alloc, heap, goroutine). Will comment back if I get any updates.

**Comment by CharlyF** [comment]
Hi all,

That is correct, the issue here is that the agent is not able to run the kubernetes state metrics check.
The best first step is to split the KSM collectors and also make use of the [Cluster Checks feature](https://docs.datadoghq.com/agent/cluster_agent/clusterchecks/) (to only have a set of 1 to 3 beefed up agent replicas that only scrape KSM endpoints). 
I can't share a precise value in terms of memory limits as it depends on the setup, but it will be close to 1Gb as you get over the 100k metrics generated by a KSM instance.

We are currently working on adding a dedicated collector in KSM directly to avoid the processing of the open metrics output. I am adding this issue to the card so we can track it internally as well.

Let me know if you need more details about the Cluster Check Setup.

Best,
.C

**Comment by dmizelle** [comment]
I may be running into a different issue then. The node my problematic agent is running on isn't the same one that has KSM. I'll wait to hear back from support and further dig down into the pprof dumps I took.

**Comment by acim** [comment]
@dmizelle, yes, still not resolved, but I am in touch with the support and will try to increase the number of runners and eventually further increase the allowed amount of memory..
